clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -300,6 +290,294 @@ private SchemaEntry fetchSingleSchema(String requestPath, boolean isValueSchema)
     return new SchemaEntry(schemaResponse.getId(), writerSchema);
   }
 
+  /**
+   * This method will try to update all value schemas.
+   * Based on Router endpoint availability, it should first try the new method: (1) Fetch all value schema IDs first,
+   * (2) then fetch each individual schema one by one. If the forceRefresh is true, it will query all known value schema
+   * IDs, otherwise, it will only query value schemas that are not populated in cached schema map.
+   * If the new router endpoint /value_schema_ids/ is not available, it should fall back to query all value schemas at once.
+   */
+  private void updateAllValueSchemas(boolean forceRefresh) {
+    try {
+      Set<Integer> valueSchemaIdSet;
+      try {
+        valueSchemaIdSet = fetchAllValueSchemaIdsFromRouter();
+      } catch (Exception e) {
+        LOGGER.warn(
+            "Caught exception when trying to fetch all value schema IDs from router, will fetch all value schema entries instead.");
```

Comment:
Do we know the difference between non-existing endpoint and some true error even the endpoint exist?
For non-existing endpoint, potentially we can skip the logging, since it is not quite useful IMO.

Reply:
I think it still throws VeniceClientHttpException with a different error message inside executeRouterRequest. I feel like it is hard to distinguish it from other query errors. Given the fact that it should not be called frequently, I suppose it is fine. What do you think?

Reply:
The only issue I can see is that the error log for the non-existing endpoint can be a little confusing.
If it is hard to differentiate, we can leave it this way.